src/main/java/org/springframework/cloud/security/oauth2/proxy/OAuth2ProxyAutoConfiguration.java

```diff
@@ -43,14 +44,24 @@
 	@Autowired
 	private ProxyAuthenticationProperties properties;
 
-	// Inject the @Primary one because it knows how to refresh an expired token 
+	// Inject the @Primary one because it knows how to refresh an expired token
 	@Autowired(required = false)
 	private OAuth2RestTemplate restTemplate;
 
+	@Autowired(required = false)
+	@LoadBalanced
+	private OAuth2RestTemplate loadBalancedRestTemplate;
+
+	@Value("${spring.oauth2.userInfo.loadBalanced:false}")
```

Comment:
Should probably be in `spring.oauth2.resource` (where the `userInfoUri` is already), i.e. in `ResourceServerProperties`.